docs/app-integration-development/08-OAuth-2-Functionality.md

```diff
@@ -51,5 +51,119 @@ There two options for implementing PagerDuty OAuth in your app. [PKCE (Proof Key
+* Decode the signature using Base65Url
+* Use the decoded signature to validate the ASCII array generated previously
+
+The contents are now validated, and can be further inspected to ensure the message is correct. The following claims can be inspected and verified;
+
+ Claim       |      Description   
+------------ | ----------------------------------
+ `acr`       |  Authentication method used should match `acr_values` request parameter
+ `iss`       |  Identifier of the token sender, and must be a plain https url
+ `sub`       |  Identifier of the user
+ `aud`       |  Expected reciever of the token which must contain a client identifier
+ `iat`       |  Time the ID Token was issued
+ `auth_time` |  Time the user was last logged in without SSO
+ `jti`       |  Unique token identifier is what was expected 
+ `nonce`     |  The nonce if passed into the token
+
+## Full ID Token Example
+
+The following is a sample of a full ID token with a valid header, payload, and signature.
+
+### Header
+
+The algorithm used to encrypt and decrypt the token is a variant of SHA-256 called HS256, and is declared as the first part of the header. The next part of the header is the Key ID, which is later used for verification. The last part is the certificate (x5t) which is base64url encoded. Putting these parts together, the header is formed.
+
+    {
+      "alg": "HS256",
+      "kid": "-38074812",
+      "x5t": "MR-pGTa866RdZLjN6Vwrfay907g"
+    }
+
+Once ecrypted, the header would form the first part of the token and would be the following;
+
+`eyJhbGciOiJIUzI1NiIsImtpZCI6Ii0zODA3NDgxMiIsIng1dCI6Ik1SLXBHVGE4NjZSZFpMak42VndyZmF5OTA3ZyJ9`
+
+### Payload
+
+The second part of the token contains all the claims sent, and in this sample case only a few claims will be made, the `iss`, `acr`, `iat`, and `exp`. The `iss` field is a sample issuer, which would be a trusted issuer sending in the token. The second part of it is the `acr` which can be validated after the signature if desired, and specifies the authentication context, which in this case is a URI. The next two fields are `iat` and `exp` which indicate the time the token was sent and its expiry time, which are highly recommended to always include for validation purposes. The final part is a public claim called `purpose` which gives the payload a label, and is completely optional.
+
+Putting all these fields together would give us a token with the following fields;
+
+    {
+      "iss": "https://www.some-issuer.com/issue_token",
+      "acr": "urn:some:acr:htmlSql",
+      "iat": 1645653858,
+      "exp": 1645655061,
+      "purpose": "sample ID token"
+    }
+
+Now encrypting the payload using the algorithm specified in the header would give the following;
+
+`eyJpc3MiOiJodHRwczovL3d3dy5zb21lLWlzc3Vlci5jb20vaXNzdWVfdG9rZW4iLCJhY3IiOiJ1cm46c29tZTphY3I6aHRtbFNxbCIsImlhdCI6MTY0NTY1Mzg1OCwiZXhwIjoxNjQ1NjU1MDYxLCJwdXJwb3NlIjoic2FtcGxlIElEIHRva2VuIn0`
+
+### Signature
```

Comment:
One thing that is missing from this section is how to get the key value pairs from pagerduty.

User's can hit the `/global/oauth/anonymous/.well-known/openid-configuration` for information about the identity server as well as get the `jwks_uri` value